rsocket/internal/InMemResumeManager.h

```diff
@@ -75,14 +115,29 @@ class InMemResumeManager : public ResumeManager {
   // Inferred position of the rcvd frames
   ResumePosition impliedPosition_{0};
 
-  // Active REQUEST_STREAMs are preserved here
-  std::set<StreamId> activeRequestStreams_;
+  // StreamIds of streams originating locally
+  std::set<StreamId> requesterRequestStreams_;
+  std::set<StreamId> requesterRequestChannels_;
+  std::set<StreamId> requesterRequestResponses_;
+
+  // StreamIds of streams originating remotely
+  std::set<StreamId> responderRequestStreams_;
+  std::set<StreamId> responderRequestChannels_;
+  std::set<StreamId> responderRequestResponses_;
+
+  // This stores the allowance for streams where the local side
+  // is a publisher (REQUEST_STREAM Responder and REQUEST_CHANNEL)
+  std::map<StreamId, uint32_t> publisherAllowance_;
+
+  // This stores the allowance for streams where the local side
+  // is a consumer (REQUEST_STREAM Requester and REQUEST_CHANNEL)
+  std::map<StreamId, uint32_t> consumerAllowance_;
 
-  // Active REQUEST_CHANNELs are preserved here
-  std::set<StreamId> activeRequestChannels_;
+  // This is a mapping between streamIds and application-aware streamTokens
+  std::map<StreamId, std::string> streamTokens_;
```

Comment:
Do we need to have 3 maps?  Can this be one `std::unordered_map<StreamId, std::tuple<uint32_t, uint32_t, std::stirng>>`?  (Or struct, whatever).